packages/jest-haste-map/src/index.ts

```diff
@@ -803,7 +803,14 @@ class HasteMap extends EventEmitter {
 
         watcher.once('ready', () => {
           clearTimeout(rejectTimeout);
-          watcher.on('all', onChange);
+
+          if (useWatchman) {
+            watcher.on('all', onChange);
+          } else {
+            (watcher as ChokidarFsWatcher).on('all', (type, filePath, stat) => {
+              onChange(type, filePath, root, stat);
+            });
```

Comment:
this is the part that makes me think creating some abstraction makes sense. There might be other things we want to tweak as well